test/local_test.go

```diff
@@ -0,0 +1,74 @@
+package test
+
+import (
+	"errors"
+	"gopkg.in/yaml.v3"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+type parsedCRD struct {
+	APIVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+	Metadata   struct {
+		Annotations struct {
+			EksAmazonawsComRoleArn              string `yaml:"eks.amazonaws.com/role-arn"`
+			EksAmazonawsComStsRegionalEndpoints string `yaml:"eks.amazonaws.com/sts-regional-endpoints"`
+		} `yaml:"annotations"`
+		Name string `yaml:"name"`
+	} `yaml:"metadata"`
+}
+
+func TestServiceAccountsUseStsRegionalEndpoint(t *testing.T) {
+	addonsPath := "../modules/cluster/addons/"
+	addonFiles, err := os.ReadDir(addonsPath)
+	if err != nil {
+		t.Error(err)
+	}
+	for _, v := range addonFiles {
+		fileName := v.Name()
+		if filepath.Ext(fileName) == ".yaml" {
+			crds, err := readManifest(addonsPath + fileName)
+			if err != nil {
+			    // do something for error handling
+			    // e.g
+			    // t.Fatalf("err=%#v\n" , err)
+			}
+			for _, v := range crds {
+				if v.Kind == "ServiceAccount" {
+					if v.Metadata.Annotations.EksAmazonawsComRoleArn != "" && v.Metadata.Annotations.EksAmazonawsComStsRegionalEndpoints != "true" {
+						t.Errorf("Service account %s role arn %s has EksAmazonawsComStsRegionalEndpoints=%s, should be 'true'", v.Metadata.Name, v.Metadata.Annotations.EksAmazonawsComRoleArn, v.Metadata.Annotations.EksAmazonawsComStsRegionalEndpoints)
```

Comment:
Consider something like
```
if v.Metadata.Annotations.EksAmazonawsComRoleArn != "" {
  assert.True(t, v.Metadata.Annotations.Eks..., fmt.Sprintf("%s should be true, ...))
}
```
Makes it a little more obvious what you're actually testing for, imo.

```